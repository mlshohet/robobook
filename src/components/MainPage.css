@font-face {
	font-display: swap;
	font-family: 'joystix';
	font-style: normal;
	font-weight: normal;
	src: local('joystix'), url('../containers/joystix.ttf') format('ttf');
}

h1 {
	font-family: 'joystix';
	color: rgb(5, 255, 189);
}