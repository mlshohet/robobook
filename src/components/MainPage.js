import React from 'react';
import Header from '../components/Header';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary';
import Clock from '../components/Clock';
import './MainPage.css';

class MainPage extends React.Component {

	constructor() {
		super();
		this.state = {
			clock: ''
		}
	}

	componentDidMount() {
		this.props.onRequestRobots();
		return fetch('https://worldtimeapi.org/api/ip')
		.then(response => response.json())
		.then(clock => {
			const datetime = clock.datetime;
			this.setState({ clock: datetime })
		}).catch(error => {
			console.log(error);
		});
	}

	filteredRobots = () => {
		return this.props.robots.filter(robot => {
			return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
		});
	}
	

	render() {
		const { onSearchChange, isPending, robots } = this.props;
		return (
			<div className = 'tc'>
				<Header />
				<SearchBox searchChange = {onSearchChange} />
					<Scroll>
						{ isPending ? <h1 className = 'tc'>LOADING...</h1>
						: <ErrorBoundary>
							<CardList robots= {this.filteredRobots()} />
							<Clock clock={this.state.clock}/>
						</ErrorBoundary>
						}
					</Scroll>
			</div>
		);
	}
}

export default MainPage;