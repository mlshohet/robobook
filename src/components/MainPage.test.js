import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {  //comes with Jest, runs before each test
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false
	}
	wrapper = shallow(<MainPage {...mockProps}/>)
});

it('renders mainPage', () => {
	expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'John',
			email: 'john@mail.com'
		}],
		searchField: 'o',
		isPending: false
	}

	const wrapper2 = shallow(<MainPage {...mockProps2}/>)


	expect(wrapper2.instance().filteredRobots([])) // instance comes with enzyme
		.toEqual(
			[{
			id: 3,
			name: 'John',
			email: 'john@mail.com'
		}]
		);

	// expect(wrapper2.instance().filteredRobots([]))
	// 	.toEqual([])
});


