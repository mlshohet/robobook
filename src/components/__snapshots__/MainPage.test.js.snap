// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mainPage 1`] = `ShallowWrapper {}`;
